ob1k-core/src/main/java/com/outbrain/ob1k/client/endpoints/AbstractClientEndpoint.java

```diff
@@ -29,13 +30,15 @@
   protected final RequestMarshallerRegistry marshallerRegistry;
   protected final HttpClient httpClient;
   public final Endpoint endpoint;
+  protected final DoubleDispatchStrategy doubleDispatchStrategy;
 
   protected AbstractClientEndpoint(final HttpClient httpClient, final RequestMarshallerRegistry marshallerRegistry,
-                                   final Endpoint endpoint) {
+                                   final Endpoint endpoint, final DoubleDispatchStrategy doubleDispatchStrategy) {
 
     this.httpClient = httpClient;
     this.marshallerRegistry = marshallerRegistry;
     this.endpoint = endpoint;
+    this.doubleDispatchStrategy = doubleDispatchStrategy;
```

Comment:
This is a nullable field - move it to a setter to simplify the c'tor, or make it always non null using a default strategy (that can be noop, but it's weird man)